src/main/kotlin/reactor/kotlin/core/publisher/FluxExtensions.kt

```diff
@@ -211,4 +211,4 @@ fun <T : Any> Flux<out Iterable<T>>.split(): Flux<T> = this.flatMapIterable { it
  * @author Kevin Davin
  * @since 3.2
  */
-fun <T> Flux<T>.switchIfEmpty(s: () -> Publisher<T>): Flux<T> = this.switchIfEmpty(Flux.defer { s() })
+fun <T> Flux<T>.switchIfEmptyDeferred(s: () -> Publisher<T>): Flux<T> = this.switchIfEmpty(Flux.defer { s() })
```

Comment:
got a better idea. Let's keep the old version (just to make sure we are doing breaking change). Add deprecation on that extension method and introduce that new extension not as a replacement but as an alternative. WDYT?

Reply:
I have a suspicion that since it was not possible to import the extension as it was always shadowed by the method from Flux, it has never been used by anyone.

Reply:
I agree, but those who did import it and never used it may end up with some compilation errors. Thus to be safe, we need to go through deprecation (fully agree with you but this is project bureaucracy we need to go through 🤦‍♂️)

Reply:
Yeah, I understand. Here's another thought: They thought they used the extension and it turns out they have a nasty hidden bug in their code (I totally did and spend a whole day debugging!), but now, after the dependency update they will see the compilation error in the import and catch the bug!

Reply:
I'm wondering if `@Deprecated` keyword provides any notable wornings in Kotlin

Reply:
done. check what to do with @since as there is a mismatch between versioning (3.2 vs current 1.1.2)

Reply:
Please add your name on it and put @since 1.1.3

Reply:
ok